homeassistant/components/compit/config_flow.py

```diff
@@ -0,0 +1,102 @@
+"""Config flow for Compit integration."""
+
+from __future__ import annotations
+
+from collections.abc import Mapping
+import logging
+from typing import Any
+
+from compit_inext_api import CannotConnect, CompitAPI, InvalidAuth, SystemInfo
+import voluptuous as vol
+
+from homeassistant.config_entries import SOURCE_REAUTH, ConfigFlow, ConfigFlowResult
+from homeassistant.const import CONF_EMAIL, CONF_PASSWORD
+from homeassistant.helpers.aiohttp_client import async_create_clientsession
+
+from .const import DOMAIN
+
+_LOGGER = logging.getLogger(__name__)
+
+STEP_USER_DATA_SCHEMA = vol.Schema(
+    {
+        vol.Required(CONF_EMAIL): str,
+        vol.Required(CONF_PASSWORD): str,
+    }
+)
+
+STEP_REAUTH_SCHEMA = vol.Schema(
+    {
+        vol.Required(CONF_PASSWORD): str,
+    }
+)
+
+
+class CompitConfigFlow(ConfigFlow, domain=DOMAIN):
+    """Handle a config flow for Compit."""
+
+    VERSION = 1
+
+    async def async_step_user(
+        self,
+        user_input: dict[str, Any] | None = None,
+    ) -> ConfigFlowResult:
+        """Handle the initial step."""
+        errors: dict[str, str] = {}
+        if user_input is not None:
+            session = async_create_clientsession(self.hass)
+            api = CompitAPI(user_input[CONF_EMAIL], user_input[CONF_PASSWORD], session)
+            system_info = None
+            try:
+                system_info = await api.authenticate()
+            except CannotConnect:
+                errors["base"] = "cannot_connect"
+            except InvalidAuth:
+                errors["base"] = "invalid_auth"
+            except Exception:
+                _LOGGER.exception("Unexpected exception")
+                errors["base"] = "unknown"
+
+            if (
+                system_info
+                and isinstance(system_info, SystemInfo)
+                and system_info.gates is not None
+            ):
```

Comment:
what if the gates is None? Currently we would not raise an error but just show the form again